vmm/src/lib.rs

```diff
@@ -2471,35 +2490,9 @@ impl RequestHandler for Vmm {
                         continue;
                     }
 
-                    match &mut socket {
-                        SocketStream::Unix(unix_socket) => {
-                            let mut buf = [0u8; 4];
-                            let (_, file) = unix_socket.recv_with_fd(&mut buf).map_err(|e| {
-                                MigratableError::MigrateReceive(anyhow!(
-                                    "Error receiving slot from socket: {}",
-                                    e
-                                ))
-                            })?;
-
-                            if existing_memory_files.is_none() {
-                                existing_memory_files = Some(HashMap::default())
-                            }
+                    existing_memory_files.push(Self::vm_receive_memory_fd(&mut socket)?);
```

Comment:
Nit: Why not just insert the result of `Self::vm_receive_memory_fd` into the given hashmap? 

I see you removed this code in the next commit so probably not relevant anyway ...